worker/src/core/loaders/message-worker.loader.ts

```diff
@@ -1,51 +1,30 @@
-import { Worker, spawn, ModuleThread } from 'threads'
-import MessageWorker from './message-worker'
+import process from 'process'
+import worker from './message-worker'
 import { loggerWithLabel } from '@core/logger'
 import config from '@core/config'
 
 const logger = loggerWithLabel(module)
 
-const createMessageWorker = async (
-  workerId: string,
-  isLogger = false
-): Promise<ModuleThread<MessageWorker>> => {
-  const worker = await spawn<MessageWorker>(new Worker('./message-worker'))
+const messageWorkerLoader = async (): Promise<void> => {
+  const workerId = String(1)
+  const isLogger = config.get('messageWorker.numLogger') > 0
+
   try {
-    await worker.init(workerId, isLogger)
+    process.on(
+      config.get('env') === 'development' ? 'SIGINT' : 'SIGTERM',
+      worker.shutdown
+    )
+    await worker.start(workerId, isLogger)
   } catch (err) {
     logger.error({
       message: 'Worker died',
       workerId,
-      isLogger,
+      isLogger: false,
```

Comment:
isLogger should be `config.get('messageWorker.numLogger') > 0`

Reply:
Fixed!